soccer_object_detection/src/soccer_object_detection/test_object_detection.py

```diff
@@ -96,3 +101,185 @@ def test_object_detection_node(self):
                 cv2.imshow("res", mat)
                 cv2.waitKey(1)
                 cv2.destroyAllWindows()
+
+    def test_visualize_annotations(self):
+        src_path = os.path.dirname(os.path.realpath(__file__))
+
+        # Data downloaded from https://github.com/bit-bots/TORSO_21_dataset
+        annotation_path = f"{src_path}/../../data/test/annotations.yaml"
+        annotation_pickle = f"{src_path}/../../data/test/annotation.pkl"
+        image_path = f"{src_path}/../../data/test/data/images/"
+
+        if not os.path.exists(annotation_pickle):
+            with open(annotation_path) as f:
+                print("Pickling annotation, will take a long time")
+                yaml_data = yaml.load(f)
+            with open(annotation_pickle, "wb") as f2:
+                pickle.dump(yaml_data, f2)
+                return
+
+        MAX_DIMENSIONS = (1778, 1000)
+
+        with open(annotation_pickle, "rb") as f:
+            annos = pickle.load(f)["images"]
+            print(
+                "Press 's' and 'd' to move between images. 'A' and 'S' let you jump 100 images.\n'c' to correct a label\n'v' to save image.\n'q' closes.\n'n' toggles not in image text. 'o' to toggle showing obstacles\n'e' to toggle all annotations"
+            )
+            files = list(annos)
+            files.sort()
+            not_in_image = True
+            show_obstacles = True
+            show_annotations = True
+            i = 0
+            while True:
+                f = files[i]
+                img_path = os.path.join(image_path, f)
+                img = cv2.imread(img_path)
+                assert img is not None
+                h, w, c = img.shape
+                text_thickness = int(w / 200)
+                line_thickness = int(w / 200)
+                y = 20
+                image_annos = annos[f]["annotations"]
+                # sort lables to have them in the correct order.
+                image_annos_sorted = []
+                correct_order = {
+                    "field edge": 0,
+                    "goalpost": 1,
+                    "left_goalpost": 2,
+                    "right_goalpost": 3,
+                    "crossbar": 4,
+                    "robot": 5,
+                    "obstacle": 6,
+                    "ball": 7,
+                    "L-Intersection": 8,
+                    "T-Intersection": 9,
+                    "X-Intersection": 10,
+                }
```

Comment:
What does this exactly mean, sort labels in the correct order? Some of these labels don't exist in our model🤔

Our model labels:
#names:
#  0: ball
#  1: goalpost
#  2: robot
#  3: L-Intersection
#  4: T-Intersection
#  5: X-Intersection
#  6: top_bar